providers/aws/connection/connection.go

```diff
@@ -172,6 +179,11 @@ func parseFlagsForConnectionOptions(m map[string]string) []ConnectionOption {
 		externalId := m["external-id"]
 		o = append(o, WithAssumeRole(cfg, role, externalId))
 	}
+
+	if len(creds) > 0 {
```

Comment:
is there a use case where there might be more creds here and we'd need to iterate over them instead of taking the first one?

Reply:
i can't think of one, since one would override the other